bin/matrix-deploy.js

```diff
@@ -205,24 +210,50 @@ Matrix.localization.init(Matrix.localesFolder, Matrix.config.locale, function ()
                       debug('DOWNLOAD URL: ' + downloadURL);
                       debug('The data sent for ' + appName + ' ( ' + appDetails.version + ' ) is: ', appData)
 
+
+                      function triggerInstall(app) { 
+                        //TODO validate content of app
+                        if (!app.hasOwnProperty('key') || !app.hasOwnProperty('data') || !app.data.hasOwnProperty('meta') || !app.data.meta.hasOwnProperty('currentVersion')) {
+                          console.error('Trying to install app missing parameters'.red);
+                          process.exit(1);
+                        }
+                        var options = {
+                          policy: policyObject,
+                          name: appName,
+                          id: app.key,
+                          versionId: app.data.meta.currentVersion
+                        }
+                        
+                        Matrix.helpers.installApp(options, function () { 
+                          endIt();
+                        });
+                      };
+
                       //Listen for the app creation in appStore
                       Matrix.firebase.appstore.watchForAppCreated(appName, function (app) {
-                        debug('App created>', app);
-                        process.exit();
+                        if (waitForRegistrationToFinish) {
+                          var deploymentTimer = setInterval(function () {
+                            if (deployFinished) { 
+                              clearTimeout(deploymentTimer);
+                              debug('App created > ' + JSON.stringify(app));
+                              triggerInstall(app);
+                            }  
+                          }, 400);
+                          
+                        } else {
+                          triggerInstall(app);
+                        }
                       });
 
-                      /*var options = {
-                        policy: result[appId].versions[versionId].policy,
-                        name: appName,
-                        id: appId,
-                        versionId: versionId
-                      }
-                      
-                      Matrix.helpers.installApp(options, function () { 
-                        endIt();
+                      /*Matrix.firebase.appstore.watchForAppUpdated(appName, function (app) {
+                        debug('App updated > ' + JSON.stringify(app));
+                        setTimeout(function () {
+                          process.nextTick(function () {
+                            triggerInstall(app);
+                          });
+                        }, 500);
```

Comment:
I'm going to say nay, it's easy to implement if we ever need that. Getting rid of them